src/GitHub.VisualStudio/UI/Views/GitHubConnectContent.xaml.cs

```diff
@@ -39,23 +41,70 @@ void signOut_Click(object sender, RoutedEventArgs e)
             signOut.IsEnabled = true;
         }
 
-        private void login_Click(object sender, RoutedEventArgs e)
+        void login_Click(object sender, RoutedEventArgs e)
         {
             login.IsEnabled = false;
             ViewModel.Login();
             login.IsEnabled = true;
         }
 
-        public GitHubConnectSection ViewModel
+        void repositories_MouseDoubleClick(object sender, System.Windows.Input.MouseButtonEventArgs e)
         {
-            get { return (GitHubConnectSection)GetValue(ViewModelProperty); }
+            ViewModel.OpenRepository();
+            e.Handled = true;
+        }
+
+        public IGitHubConnectSection ViewModel
+        {
+            get { return (IGitHubConnectSection)GetValue(ViewModelProperty); }
             set { SetValue(ViewModelProperty, value); }
         }
 
         public static readonly DependencyProperty ViewModelProperty =
             DependencyProperty.Register(
                 "ViewModel",
-                typeof(GitHubConnectSection),
+                typeof(IGitHubConnectSection),
                 typeof(GitHubConnectContent));
     }
+
+    public class FormatRepositoryNameConverter : IMultiValueConverter
+    {
+        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (values.Length != 2 || parameter as string != "FormatRepositoryName")
+                return String.Empty;
+
+            var item = values[0] as ISimpleRepositoryModel;
+            var context = values[1] as IGitHubConnectSection;
+            if (context == null)
+                return item?.Name ?? String.Empty;
+            
+            if (context.SectionConnection.Username == item.CloneUrl.Owner)
```

Comment:
If `context` not null, but `item` was null, we'd get a `NullReferenceException` here. Perhaps line 79 should be `if (context == null || item == null)`